### PR TITLE
Adding Sell product and Sell-related locations.

### PR DESCRIPTION
**Pull request by grzegorz-dyk**
These locations will be used firstly by Sell app market. Sell-specific
apps will be able to determine in which Sell location they will appear
at.

This is part of Sell's App Framework project.

### CODE REVIEW

File: lib/zendesk_apps_support/product.rb [hunk 1]
```diff
@@ -19,11 +19,13 @@ def self.all
     PRODUCTS_AVAILABLE = [
       Product.new(code: 1, name: 'support', legacy_name: 'zendesk'),
       Product.new(code: 2, name: 'chat', legacy_name: 'zopim'),
-      Product.new(code: 3, name: 'standalone_chat', legacy_name: 'lotus_box')
+      Product.new(code: 3, name: 'standalone_chat', legacy_name: 'lotus_box'),
+      Product.new(code: 4, name: 'sell', legacy_name: 'base')
```
Comment (grzegorz-dyk) [inline lib/zendesk_apps_support/product.rb]: @anyone Do I need to populate these in the DB? If yes, how can I do it?

Comment (grzegorz-dyk) [reply lib/zendesk_apps_support/product.rb]: Also, not sure If Sell needs `legacy_name`? We did not appear under legacy name in Zendesk Suite...

Comment (bbrzoska) [reply lib/zendesk_apps_support/product.rb]: @grzegorz-dyk I see `legacy_name` used here: https://github.com/zendesk/zendesk_app_market/blob/9036852175a27ddb40a1d97d9b8861951053df18/app/models/location.rb#L19

we could use the .name if .legacy_name doesn't exist in `app_market` or simply define `sell` in both cases.

File: lib/zendesk_apps_support/location.rb [hunk 2]
```diff
@@ -49,7 +49,17 @@ def self.all
       Location.new(id: 13, name: 'system_top_bar',
                    product_code: Product::STANDALONE_CHAT.code, v2_only: false),
       Location.new(id: 14, name: 'background',
-                   product_code: Product::CHAT.code)
+                   product_code: Product::CHAT.code),
+      Location.new(id: 15, name: 'top_bar', product_code: Product::SELL.code),
```
Comment (Mehul-Porwal) [inline lib/zendesk_apps_support/location.rb]: Is Sell really building so many locations?

Comment (grzegorz-dyk) [reply lib/zendesk_apps_support/location.rb]: @Mehul-Porwal We do have these in frontend. I was advised to add these typical ones (`nav_bar`, `top_bar` etc).

Comment (bbrzoska) [reply lib/zendesk_apps_support/location.rb]: @Mehul-Porwal we're launching with 4 location at start, but we'll be adding others listed sometime next year. If it's a relatively easy change to add new locations in the DB, we can start with 4 and take it from there.

Comment (Mehul-Porwal) [reply lib/zendesk_apps_support/location.rb]: You do not need to add it to DB... just adding here is all that is required. Maybe just add the 4 you need at the moment and add more when required.

Comment (bbrzoska) [reply lib/zendesk_apps_support/location.rb]: I see, thanks @Mehul-Porwal. Why is the DB mentioned here though?

Comment (grzegorz-dyk) [reply lib/zendesk_apps_support/location.rb]: @bbrzoska I asked whether I should add them to DB in other comment. It did not get attached to thread though :/

Comment (Mehul-Porwal) [reply lib/zendesk_apps_support/location.rb]: The ids defined here are used in DB, when someone installs an app for 1Location.new(id: 17, name: 'deal_card', product_code: Product::SELL.code, collapsible: true),` we will use the code `17` in DB for the app_location, which will then we used to link to the definition here. Which means once in production, you can never change the id codes.

Comment (bbrzoska) [reply lib/zendesk_apps_support/location.rb]: Ah, that makes total sense. Thanks for the explanation @Mehul-Porwal!

**Pull request merged**